### PR TITLE
docs: add new doc that describes plugin advantages

### PR DESCRIPTION
**Pull request by andrew-hirata-playeveryware**

### CODE REVIEW

**Comment by scovertb** [comment]
Should this not be at the head of the main readme, since it's the first thing people might see when looking into the plugin?

**Comment by andrew-hirata-playeveryware** [comment]
> Should this not be at the head of the main readme, since it's the first thing people might see when looking into the plugin?

The reason it's in a separate document is so that the main readme doesn't get "too long".  @RedMazer might also have some opinions about my logic in making it a separate document too.

**Comment by scovertb** [comment]
In that case a link to this doc should be added to the top of the main readme.

File: docs/plugin_advantages.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# Advantages of using this Plugin
+
+## Handles Windows Init
+One of the core features of the Unity Plugin for EOS, is that it handles
+doing the init code in such a way that the Overlay will work correctly in 
+standalone builds on windows. Due to how the EOS SDK does it's graphics initialization
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Windows should be capitalized here and in the rest of the doc.

File: docs/plugin_advantages.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+standalone builds on windows. Due to how the EOS SDK does it's graphics initialization 
+for the overlay, the code that call's into the EOS SDK for init needs to happen _before_
+the Unity finishes doing it's graphics init code. The only place to do this, is in native
+code, in a specially named DLL that gets called before the engine finishes running. 
+If one is using this Plugin, one doesn't have to worry about any of that, as it's handled 
+automatically.
+
+
+## Handles the Unity Editor
+Need to unload and reload the EOS SDK dynamic libraries to ensure new config works
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Line 15: Is this a directive or an explanation of functionality? Supposed to be a complete sentence?

File: docs/plugin_advantages.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+this would mean that any changes that need to be done to the EOS configuration would require an
+editor reboot.
+
+To resolve that, we have written additional code to handle loading, and unloading, the EOS SDK manually.
+This both ensures that new configuration changes can be made without restarting the editor, and makes sure
+that any callbacks from previous play sessions aren't called, and that all EOS objects are destroyed between 
+runs in the Editor.
+
+## Provides Samples
+One feature of the Plugin that particularly makes it standout from using just the C# SDK that comes from Epic,
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Phrasing "just the C# SDK that comes from Epic" might annoy someone.

File: docs/plugin_advantages.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+editor reboot.
+
+To resolve that, we have written additional code to handle loading, and unloading, the EOS SDK manually.
+This both ensures that new configuration changes can be made without restarting the editor, and makes sure
+that any callbacks from previous play sessions aren't called, and that all EOS objects are destroyed between 
+runs in the Editor.
+
+## Provides Samples
+One feature of the Plugin that particularly makes it standout from using just the C# SDK that comes from Epic, 
+are the Unity specific samples that showcase both how to use the EOS SDK, and provide copy-past-able code
```
Comment (scovertb) [inline docs/plugin_advantages.md]: "copy-past-able" should probably just be "copyable."

File: docs/plugin_advantages.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+that can be used in projects to give one a head-start on integrating the EOS SDK. Because they're provided as UPM
+samples, they can be imported, deleted, and modified as needed to allow users of the Plugin maximum flexibility.
+
+## Works with Unity
+In order to facilitate configuration of the EOS SDK, the Plugin comes with editor scripts that ensure 
+proper inclusion of EOS SDK binaries and configuration files into the final standalone build. 
+Also included in the Plugin, are tools to store and load the configuration for the EOS SDK so that it can
+initialize correctly, and for setting up EAC (Epic Anti-cheat).
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Line 38: EAC stands for Easy Anti-Cheat.

File: docs/plugin_advantages.md [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+
+## Provides integration with NetCode for game objects
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Exact formatting of the plugin name is "Netcode for GameObjects"

File: docs/plugin_advantages.md [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+Provided as a part of the samples, the Plugin provides Transport layer that lets one use EOS with Netcode for game objects.
```
Comment (scovertb) [inline docs/plugin_advantages.md]: "provides Transport layer" should probably be "implements a sample NetworkTransport"

File: docs/plugin_advantages.md [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+By using the provided sample, and by editing as needed for a specific title, one can more easily switch from a different 
+backend to EOS when using Netcode for game objects.
+
+## Comes with documentation
+The UPM and the repo come with additional documentation that can supplement the first-party EOS documentation. In particular,
+it includes resources that are relevant to running the samples, such as in sample readme files, tooltips, and documents that
+answer common questions, and explain the _why_ behind some of the choices made in the Plugin.
+
+## comes with the EOS SDK
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Line 50: Capitalize

File: docs/plugin_advantages.md [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+Each release of the Plugin comes with the correct EOS SDK, ensuring that Plugin works with the EOS SDK
```
Comment (scovertb) [inline docs/plugin_advantages.md]: Line 51: Missing period

**Review by scovertb — CHANGES_REQUESTED** [review]

File: docs/plugin_advantages.md (hunk 2 above)
Comment (andrew-hirata-playeveryware) [reply docs/plugin_advantages.md]: That is a line that needs to be deleted: thank you!

File: docs/plugin_advantages.md (hunk 3 above)
Comment (andrew-hirata-playeveryware) [reply docs/plugin_advantages.md]: That's a valid point: I do need some better phrasing. I can be less direct with my wording.

**Review by scovertb — APPROVED** [review]

**Comment by RedMazer** [comment]
for advantages, Provides Samples might be better as Provides Samples and Manager Scripts.
also a few things worth noting as well,
our level of overlay support across platforms (no idea what the process is for getting that working on other platforms.
and
our custom tools

**Review by RedMazer — CHANGES_REQUESTED** [review]
^ comment above

File: docs/plugin_advantages.md [hunk 10]
```diff
@@ -0,0 +1,50 @@
+# Advantages of using this Plugin
+
+## Handles Windows Init
+One of the core features of the Unity Plugin for EOS, is that it handles
+doing the init code in such a way that the Overlay will work correctly in 
+standalone builds on Windows. Due to how the EOS SDK does it's graphics initialization
```
Comment (arthur740212) [inline docs/plugin_advantages.md]: Line 6 ~ 8   The apostrophes could be removed

**Comment by andrew-hirata-playeveryware** [comment]
Updated with new changes  @RedMazer

**Review by RedMazer — APPROVED** [review]

**Review by Raxnam — APPROVED** [review]
with the changes it looks good to me

**Review by arthur740212 — APPROVED** [review]

**Pull request merged**